cinema/views.py

```diff
@@ -56,3 +92,59 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if self.action == "list":
+            queryset = (
+                queryset.
+                select_related("movie", "cinema_hall").
+                annotate(tickets_available=(F("cinema_hall__rows")
+                                            * F("cinema_hall__seats_in_row")
+                                            - Count("tickets"))
+                         )
+            )
+
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            queryset = queryset.filter(movie_id=int(movie))
```

Comment:
no need to use int here

Reply:
I thought that we are receiving value as string, because in another filter features we used ids to int func, but I`ll fix it)